lib/programmemory.cpp

```diff
@@ -1327,7 +1327,8 @@ namespace {
         ValueFlow::Value executeMultiCondition(bool b, const Token* expr)
         {
             if (pm->hasValue(expr->exprId())) {
-                const ValueFlow::Value& v = pm->at(expr->exprId());
+                const auto& pm2 = *pm;
+                const ValueFlow::Value& v = pm2.at(expr->exprId());
```

Comment:
Can we just remove the non-const overload for `at`? I dont think its ever used.

Reply:
It is used in three places in `Executor::executeImpl()`.

Reply:
Maybe we can rename the mutable version as `mutate_at` to make it explicit when we want to mutate.

Reply:
I think that is unnecessary. I will add a barebones `as_const()` to clean up the code.